interview-questions.md

```diff
@@ -5,50 +5,81 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 3. What is the difference between a class and an object?
 
   Your answer:
+  A class is kind of like a template child to the parent object, allowing it to define the properties of the parent.
 
   Researched answer:
-
+  "Classes and objects from the essential part of Object-oriented programming, where a class can be considered as a construct that encapsulates a group of variables and methods; whereas, an object acts as member or instance of that class."
 
 
 4. What is the difference between a HTML div and a span?
  
   Your answer:
+  Generally, a div tag is used for layout placement and containing large blocks of data. 
+  Span is used for smaller data, much like a paragraph tag or h1 that will only hold a small snippit of data.
 
   Researched answer:
+  "The difference between span and div is that a span element is in-line and usually used for a small chunk of HTML inside a line (such as inside a paragraph) whereas a div (division) element is block-line (which is basically equivalent to having a line-break before and after it) and used to group larger chunks of code."
 
 5. What is React.Fragment and why would you want to use it?
   
   Your answer:
+  Honestly, I'm drawing a blank. I dont think we covered it a whole lot and ontop of that, I dont feel secure in my React knowledge at all. Error 500. Proceeding to reteach myself in my research and extended study of React. 
   
   Researched answer:
+  "A common pattern in React is for a component to return multiple elements. Fragments let you group a list of children without adding extra nodes to the DOM."
+  While my researched study on this produced the above answer straight from React's website, I don't feel like this is really taking hold with me, but thats due to my unease with React. I'm currently taking an extra Udemy and a Codecademy course on React to suppliment my study.  
+  
 
 6. What are three options for creating responsive design?
 
   Your answer:
-
+  1: Media queries.
+  2: Using css class atributes such as flex, and margin. 
+  3: Individual stylesheets that can be loaded upon identification of device type.
+  
   Researched answer:
+  "Responsive website design consists of the following three main components:
+  Flexible layouts - Using a flexible grid to create the website layout that will dynamically resize to any width.
+  Media queries - An extension to media types when targeting and including styles. Media queries allow designers to specify different styles for specific browser and device circumstances.
+  Flexible media - Makes media (images, video and other formats) scalable, by changing the size of the media as the size of the viewport changes."
 
 
 
 ## Looking Ahead: Terms for Next Week
 - Object oriented programming
+  A type of programming that allows developers to define data type of a specific data structure, along with its functions. 
+  Its.. kind of hard to explain in just a sentance or two. 
```

Comment:
true. very hard to explain in a sentence or two. it's all very conceptual until you get to practice with ruby specifically.